pkg/aspect/metricsManager.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2017 The Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aspect
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/genproto/googleapis/rpc/code"
+
+	"istio.io/mixer/pkg/adapter"
+	aconfig "istio.io/mixer/pkg/aspect/config"
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/config"
+	"istio.io/mixer/pkg/expr"
+
+	dpb "istio.io/api/mixer/v1/config/descriptor"
+)
+
+type (
+	metricsManager struct {
+	}
+
+	metricInfo struct {
+		metricKind adapter.MetricKind
+		value      string
+		labels     map[string]string
+	}
+
+	metricsWrapper struct {
+		aspect   adapter.MetricsAspect
+		metadata map[string]*metricInfo // metric name -> info
+	}
+)
+
+// NewMetricsManager returns a manager for the metric aspect.
+func NewMetricsManager() Manager {
+	return &metricsManager{}
+}
+
+// NewAspect creates a metric aspect.
+func (m *metricsManager) NewAspect(c *config.Combined, a adapter.Builder, env adapter.Env) (Wrapper, error) {
+	params := c.Aspect.Params.(*aconfig.MetricsParams)
+
+	// TODO: get descriptors from config
+	// TODO: sync these schemas with the new standardized metric schemas.
+	desc := []*dpb.MetricDescriptor{
+		{
+			Name:  "request_count",
+			Kind:  dpb.COUNTER,
+			Value: dpb.INT64,
+			Labels: []*dpb.LabelDescriptor{
+				{Name: "source", ValueType: dpb.STRING},
+				{Name: "target", ValueType: dpb.STRING},
+				{Name: "service", ValueType: dpb.STRING},
+			},
+		},
+		{
+			Name:  "error_count",
```

Comment:
OK. @ZackButcher, let's remove the error_count metric and add a `response_code` field to `request_count`.